src/compression/b44/mod.rs

```diff
@@ -0,0 +1,863 @@
+                sample_type: SampleType::F16,
+                name: Default::default(),
+                quantize_linearly: false,
+                sampling: Vec2(1, 1),
+            };
+
+            // Two similar channels.
+            let channels = ChannelList::new(smallvec![channel.clone(), channel]);
+
+            let rectangle = IntegerBounds {
+                position: Vec2(-30, 100),
+                size: Vec2(322, 731),
+            };
+            // let rectangle = IntegerBounds {
+            //     position: Vec2(-1, 3),
+            //     size: Vec2(10, 10),
+            // };
+
+            test_roundtrip_noise_with(channels, rectangle);
+    }
+
+    #[test]
+    fn roundtrip_any_sample_type_f32() {
+            let channel = ChannelDescription {
+                sample_type: SampleType::F32,
+                name: Default::default(),
+                quantize_linearly: false,
+                sampling: Vec2(1, 1),
+            };
+
+            // Two similar channels.
+            let channels = ChannelList::new(smallvec![channel.clone(), channel]);
+
+            let rectangle = IntegerBounds {
+                position: Vec2(-30, 100),
+                size: Vec2(322, 731),
+            };
+
+            test_roundtrip_noise_with(channels, rectangle);
+    }
+
+    #[test]
+    fn roundtrip_any_sample_type_u32() {
+            let channel = ChannelDescription {
+                sample_type: SampleType::U32,
+                name: Default::default(),
+                quantize_linearly: false,
+                sampling: Vec2(1, 1),
+            };
+
+            // Two similar channels.
+            let channels = ChannelList::new(smallvec![channel.clone(), channel]);
+
+            let rectangle = IntegerBounds {
+                position: Vec2(-30, 100),
+                size: Vec2(322, 731),
+            };
+
+            test_roundtrip_noise_with(channels, rectangle);
+    }
+
+    #[test]
+    fn roundtrip_any_sample_type_mix_f32_f16_u32() {
+
+        let channels = ChannelList::new(smallvec![
+            ChannelDescription {
+                sample_type: SampleType::F32,
+                name: Default::default(),
+                quantize_linearly: false,
+                sampling: Vec2(1, 1),
+            },
+            ChannelDescription {
+                sample_type: SampleType::F16,
+                name: Default::default(),
+                quantize_linearly: false,
+                sampling: Vec2(1, 1),
+            },
+            ChannelDescription {
+                sample_type: SampleType::U32,
+                name: Default::default(),
+                quantize_linearly: false,
+                sampling: Vec2(1, 1),
+            }
+        ]);
+
+        let rectangle = IntegerBounds {
+            position: Vec2(-30, 100),
+            size: Vec2(322, 731),
+        };
+
+        test_roundtrip_noise_with(channels, rectangle);
+    }
+
+    #[test]
+    fn roundtrip_any_sample_type_toto() {
```

Comment:
Please don't use absolute paths in the tests. I suggest you either remove this test, or write to an in-memory byte vec instead of a file.

Reply:
This one will disappear.